dev/ci/docker/bionic_coq/Dockerfile

```diff
@@ -19,15 +19,19 @@ ENV NJOBS="2" \
     OPAMROOTISOK="true"
 
 # Base opam is the set of base packages required by Coq
-ENV COMPILER="4.02.3" \
-    BASE_OPAM="num ocamlfind jbuilder ounit"
+ENV COMPILER="4.02.3"
 
 RUN opam init -a -y -j $NJOBS --compiler="$COMPILER" default https://opam.ocaml.org && eval $(opam config env) && opam update
 
-# Setup of the base switch; CI_OPAM contains Coq's CI dependencies.
+# Common OPAM packages.
+# `num` does not have a version number as the right version to install varies
+# with the compiler version.
+ENV BASE_OPAM="num ocamlfind.1.8.0 jbuilder.1.0+beta20 ounit.2.0.8" \
+    CI_OPAM="menhir.20180528 elpi.1.0.3 ocamlgraph.1.8.8"
```

Comment:
I don't think this is a good idea.

IMO we should ask the upstreams of our CI module X to provide an opam file (when they have external dependencies) and then run `opam install --deps-only X` instead of hard coding the dependencies here.

Otherwise I (with the upstream hat) I've an hard time to update `coq-elpi`. It is no more enough to move the `coq-master` branch Coq-CI is tracking, but I also have to make a PR for this file, and I don't even know it exists.

Reply:
The docker based system saves us a lot of build time but doesn't allow the solution that you propose. Anyways, you shouldn't have to worry whenever you update your dependencies. I plan to implement commit pinning soon as when we have that, this problem will go away.

Reply:
Not that your request for changes is not relevant as this PR doesn't change the model. In the previous situation, we also used a fixed set of dependencies but we didn't know which ones exactly.

Reply:
> The docker based system saves us a lot of build time but doesn't allow the solution that you propose

I don't get it. Exactly as you run `opam install elpi.XXX` in the script above, you could run `opam install coq-elpi --deps-only`. Then as soon as the docker image is generated  the dependencies of coq-elpi are going to get right.

> Not that your request for changes is not relevant as this PR doesn't change the model.

Maybe what I'm missing is *when* the image is generated. If one need to change this file in oder to have it re-generated then you can discard my review.

Reply:
Regeneration is on every build unless you set a variable to cache the image, which we do.

We could do what you propose but I am not sure I like the non-determinism there; there is always the tradeoff with caching; what you are telling us is that we shouldn't cache elpi, indeed, we could have a more complex logic to be smarter in caching, but I think for now the tradeoff is acceptable.

Reply:
Also note that docker tends not to like non-deterministic commands, IIUC.

Reply:
So, in other words, today our Docker cache invalidation is manual, and for the moment it will remain so, thus depending packages must manually signal a cache invalidation attempt or override the status in their `ci-` script. The same was true before this PR, but it now makes explicit the values used in the cached image, which is a good thing "(TM)"